aten/src/ATen/THSizeStrideCompat.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <ATen/ScalarType.h>
+#include <vector>
+
+// NOTE: these functions are for compatibility into TH functions that takes sizes and strides.
+// We should just write the TH functions that don't require this, but that involves two steps:
+// 1) first class scalar support (for sizes)
+// 2) differentiating between nullptr/non-nullptr strides (the former "infers" strides).
+
+namespace at {
+
+static inline std::vector<int64_t> get_intlist_size_th(IntList sizes) {
+  if (sizes.size() == 0) {
```

Comment:
If you wanted to make this return an `IntList` you could do:

```
static int64_t one = 1;
return IntList(one);
```

Reply:
Seems like a good idea, I'll do it when I can get rid of the stacked diff.